src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -89,7 +89,7 @@
     private Map<String, String> detectorTasks;
 
     // Use this field to cache all HC tasks. Key is detector id
-    private Map<String, ADHCBatchTaskCache> hcTaskCaches;
+    private Map<String, ADHCBatchTaskCache> hcBatchTaskCaches;
```

Comment:
What are the differences between hcBatchTaskCaches and batchTaskCaches? Do they have overlap contents? Is it that hcBatchTaskCaches have detector level task cache, while batchTaskCaches has entity level task cache (if it is a HC detector) and detector level task cache (if it is single-stream detector)? if yes, then I am more confused. Why cannot we combine these two maps?

Reply:
Your understanding is correct. `hcBatchTaskCaches`: cache HC detector level information on coordinating node, will add more comments. I think the comment for `batchTaskCaches` is enough, you can read and suggest more if the comment is still not clear.

They don't have overlap contents. Combine these two maps will mix the coordinating node and worker node cache together. I don't prefer that way now. If no big concern or cons , how about we keep current design now?